### PR TITLE
    feat(core): drop support for zone.js 0.10.x     

### PR DESCRIPTION
**Pull request by alan-agius4**
With this change we drop support for zone.js 0.10.x.
This is needed because in version 12 the CLI will only work with `~0.11.4`. See angular/angular-cli#20034.

BREAKING CHANGE:

Minimum supported `zone.js` version is `0.11.4`

### CODE REVIEW

**Review by gkalpak — CHANGES_REQUESTED** [review]
Yay :tada:

I still see some occurrences of `zone.js/dist/...` (mostly in error messages and docs. They should also be updated, right?

- [aio/content/guide/testing-components-scenarios.md](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/aio/content/guide/testing-components-scenarios.md)
- [dev-infra/BUILD.bazel](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/dev-infra/BUILD.bazel)
- [packages/core/src/testability/testability.ts](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/core/src/testability/testability.ts)
- [packages/core/testing/src/async.ts](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/core/testing/src/async.ts)
- [packages/localize/schematics/ng-add/index_spec.ts](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/localize/schematics/ng-add/index_spec.ts)
- [packages/zone.js/MODULE.md](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/zone.js/MODULE.md)
- [packages/zone.js/NON_STANDARD_APIS.md](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/zone.js/NON_STANDARD_APIS.md)
- [packages/zone.js/README.md](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/zone.js/README.md)
- [packages/zone.js/lib/zone-spec/async-test.ts](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/zone.js/lib/zone-spec/async-test.ts)
- [packages/zone.js/lib/zone-spec/fake-async-test.ts](https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/packages/zone.js/lib/zone-spec/fake-async-test.ts)


Also, I think we should add the `zone.js/plugins/` directory in `.github/angular-robot.yml`:

https://github.com/angular/angular/blob/838e3bdb6d97bb3251b5518febc1fa9d110e3c4b/.github/angular-robot.yml#L76-L80

I accept that some (or maybe all) of these changes are not directly related to this PR (and should have happened independently). I am bringing them up as you seem to have made similar changes in other places as part of this PR, but happy to captrure them in an issue to make them in a new PR (also happy to take them on).

##
BTW, there are some minor typos in the commit messages:

- One of the commits says _"Minimum supported `zone.js` version is `0.11.1`"_, but based on the changes in `packages/core/package.json`, it seems that `0.11.3` is the minimum required version. Do I miss something?
- ```diff
  -This is needed because in version 12. The CLI will only work with 0.11.x. angular/angular-cli#20034
  +This is needed because in version 12 the CLI will only work with 0.11.x. See angular/angular-cli#20034.
  ```
- ```diff
  -In CLI version 12, the old style of imports are no longer supported.
  +In CLI version 12, the old style of imports is no longer supported.
  ```
- `zone.js` incosistently appearing both wrapped in backticks and unwrapped :grin:
- Some trailing periods are missing :grin:

File: aio/tools/examples/shared/yarn.lock [hunk 1]
```diff
@@ -3508,7 +3508,7 @@ codelyzer@^6.0.0:
     source-map "^0.5.7"
     sprintf-js "^1.1.2"
     tslib "^1.10.0"
-    zone.js "~0.10.3"
+    zone.js "~0.11.3"
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply aio/tools/examples/shared/yarn.lock]: Can you elaborate why this is incorrect?

File: aio/tools/examples/shared/yarn.lock [hunk 2]
```diff
@@ -12659,7 +12659,7 @@ yn@^3.0.0:
   resolved "https://registry.yarnpkg.com/yn/-/yn-3.1.1.tgz#1e87401a09d767c1d5eab26a6e4c185182d2eb50"
   integrity sha512-Ux4ygGWsu2c7isFWe8Yu1YluJmqVhxqK2cLXNQA5AcC3QfbGNpM7fu0Y8b/z16pXLnFxZYvWhd3fhBY9DLmC6Q==
 
-zone.js@~0.10.3:
+zone.js@~0.11.3:
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply aio/tools/examples/shared/yarn.lock:12662]: This is auto generated.

File: aio/tools/examples/shared/yarn.lock [hunk 3]
```diff
@@ -12660,6 +12660,8 @@ yn@^3.0.0:
   integrity sha512-Ux4ygGWsu2c7isFWe8Yu1YluJmqVhxqK2cLXNQA5AcC3QfbGNpM7fu0Y8b/z16pXLnFxZYvWhd3fhBY9DLmC6Q==
 
 zone.js@~0.11.3:
-  version "0.10.3"
-  resolved "https://registry.yarnpkg.com/zone.js/-/zone.js-0.10.3.tgz#3e5e4da03c607c9dcd92e37dd35687a14a140c16"
-  integrity sha512-LXVLVEq0NNOqK/fLJo3d0kfzd4sxwn2/h67/02pjCjfKDxgx1i9QqpvtHD8CrBnSSwMw5+dy11O7FRX5mkO7Cg==
+  version "0.11.3"
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply aio/tools/examples/shared/yarn.lock:12663]: I am not sure what you mean by we need both versions. This lock file is auto generated.

File: aio/tools/examples/shared/yarn.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (alan-agius4) [reply aio/tools/examples/shared/yarn.lock:12662]: Oh I see the problem now.

**Comment by alan-agius4** [comment]
Thanks for the review @gkalpak. I can definitely that the other suggest changes.

Let me get back to this PR on Monday.

**Comment by JiaLiPassion** [comment]
@alan-agius4 , since `zone.js` 0.11.1~0.11.3 has an issue with `es2015` bundle, and it has been fixed here https://github.com/angular/angular/pull/40456, and is available in `0.11.4`. This issue will have some impact when user using the new bundle `import 'zone.js'` , So I think maybe we should use the new `0.11.4` version as the minimum support version?

**Comment by alan-agius4** [comment]
@JiaLiPassion, that would be possible, when is 0.11.4 scheduled to be released?

**Comment by JiaLiPassion** [comment]
@alan-agius4 , the release PR has been merged, I believe it will be released very soon.

**Comment by alan-agius4** [comment]
@JiaLiPassion & @gkalpak, I update the PR once `zone.js` 0.11.4 is released I'll update the PR with the new version. 

@JiaLiPassion, can we prioritize a zone.js release? Since this will also need to be synced in https://github.com/angular/angular-cli/pull/20034

**Comment by JiaLiPassion** [comment]
@alan-agius4 , sure, I will contact @mhevery to make the new release.

**Review by gkalpak — APPROVED** [review]
Awesome work, @alan-agius4! Very thorough :100:

One last thing:
Should we update the reference to Angular CLI in [packages/zone.js/README.md (line 32)][1]?

```diff
-If you are using `Angular CLI`, the `polyfills.ts` file will contain:
+If you are using an older `Angular CLI`, the `polyfills.ts` file will contain:
```
(...or something along those lines.)

[1]: https://github.com/angular/angular/blame/618bb8bf267c5d0e00b684705b5dd812469e2df5/packages/zone.js/README.md#L32

**Comment by JiaLiPassion** [comment]
@alan-agius4 , zone.js 0.11.4 is released, please update to this new version.

**Review by JiaLiPassion — APPROVED** [review]
LGTM after update to 0.11.4

**Review by AndrewKushnir — APPROVED** [review]

**Comment by alan-agius4** [comment]
Updated PR to use zone.js 0.11.4.

File: .github/angular-robot.yml [hunk 4]
```diff
@@ -77,6 +77,7 @@ merge:
       - "packages/zone.js/dist/**"
       - "packages/zone.js/doc/**"
       - "packages/zone.js/example/**"
+      - "packages/zone.js/plugins/**"
```
Comment (josephperrott) [inline .github/angular-robot.yml:80]: This code should not be excluded as it is synced into google3.

**Review by josephperrott — CHANGES_REQUESTED** [review]

File: .github/angular-robot.yml (hunk 4 above)
Comment (alan-agius4) [reply .github/angular-robot.yml:80]: PTAL

**Review by josephperrott — APPROVED** [review]
LGTM

**Review by alxhub — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:359064352:BASE:359064387:1614101425642:90460e4f).

**Pull request closed without merging**